File: Longest Palindromic Substring/Longest_Palindromic_Substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        start, length = 0, 0
        n = len(s)
        for i in range(n):
            left, right = i, i
            while left >= 0 and right < n and s[left] == s[right]:
                if right - left + 1 > length:
                    length = right - left + 1
                    start = left
                left -= 1
                right += 1

            left, right = i, i + 1
            while left >= 0 and right < n and s[left] == s[right]:
                if right - left + 1 > length:
                    length = right - left + 1
                    start = left
                left -= 1
                right += 1
                
        return s[start:start + length]

File: Longest Palindromic Substring/Longest_Palindromic_Substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int start = 0;
        int len = 0;
        int n = s.size();
        for(int i = 0; i<n; i++) {
            int left = i, right = i;
            // for odd length palindrome
            while(left >= 0 && right < n && s[left] == s[right]) {
                if(right-left+1 > len) {
                    len = right-left+1;
                    start = left;
                }
                left--;
                right++;
            }
            // for even length palindrome
            left = i,right = i+1;
            while(left >=0 && right < n && s[left] == s[right]) {
                if(right-left+1 > len) {
                    len = right-left+1;
                    start = left;
                }
                left--;
                right++;
            }
            
        }
        return s.substr(start,len);
    }
};